e3sm_diags/driver/qbo_driver.py

```diff
@@ -341,3 +355,67 @@ def deseason(xraw):
             # i.e., get the difference between this month's value and it's "usual" value
             x_deseasoned[month_index] = xraw[month_index] - xclim[month]
     return x_deseasoned
+
+
+def _calculate_wavelet(var: xr.DataArray) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Calculate the wavelet spectrum for a given data array at a specified power
+    spectral level.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+        The variable data.
+
+    Returns
+    -------
+    Tuple[np.ndarray, np.ndarray]
+        The wavelet period and wavelet array.
+    """
+    # Find the closest value for power spectral level in the list
+    test_lev = xc.get_dim_coords(var, axis="Z")
+    test_lev_list = list(test_lev)
+    closest_lev = min(test_lev_list, key=lambda x: abs(x - POW_SPEC_LEV))
+    closest_index = test_lev_list.index(closest_lev)
+
+    # Grab target vertical level
+    data_avg = var.values[:, closest_index]
+
+    # Convert to anomalies
+    data_avg = data_avg - data_avg.mean()
+
+    # Detrend the data
+    detrended_data = detrend(data_avg)
+
+    wave_period, wavelet = _get_psd_from_wavelet(detrended_data)
+
+    # Get square root values of wavelet spectra
+    wavelet = np.sqrt(wavelet)
+
+    return wave_period, wavelet
```

Comment:
Extracted this private function since the logic is the same for test and ref (look [here](https://github.com/E3SM-Project/e3sm_diags/pull/836/files)).

Thanks GitHub Co-Pilot.